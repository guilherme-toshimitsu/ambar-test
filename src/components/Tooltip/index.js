export { default } from './Tooltip';
export { default as InfoIconTooltip } from './InfoIconTooltip';
