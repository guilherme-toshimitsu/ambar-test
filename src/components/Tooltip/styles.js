const styles = {
  tooltip: {
    fontSize: 12,
    letterSpacing: 0.2,
    wordWrap: 'break-word',
  },
  arrow: {},
};

export default styles;
