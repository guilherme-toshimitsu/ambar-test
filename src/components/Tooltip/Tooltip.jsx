import React, { memo } from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import { Tooltip as MuiTooltip, makeStyles } from '@material-ui/core';

import styles from './styles';

const useStyles = makeStyles(styles);

const Tooltip = ({ title, children, placement, classes: customClasses }) => {
  const classes = useStyles();
  const tooltipStyles = clsx(classes.tooltip, customClasses.tooltip);
  const arrowStyles = clsx(classes.arrow, customClasses.arrow);

  return (
    <MuiTooltip title={title} placement={placement} arrow classes={{ tooltip: tooltipStyles, arrow: arrowStyles }}>
      {children}
    </MuiTooltip>
  );
};

Tooltip.defaultProps = {
  title: '',
  placement: 'top',
  classes: {},
};

Tooltip.propTypes = {
  title: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node), PropTypes.string]),
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]).isRequired,
  placement: PropTypes.string,
  classes: PropTypes.object,
};

export default memo(Tooltip);
