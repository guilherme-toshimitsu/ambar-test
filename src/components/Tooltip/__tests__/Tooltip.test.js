import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Tooltip from '../';

describe('<Home> Home Page Container', () => {
  it('renders', () => {
    const container = render(
      <Tooltip title={'teste'}>
        <div>oi</div>
      </Tooltip>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
