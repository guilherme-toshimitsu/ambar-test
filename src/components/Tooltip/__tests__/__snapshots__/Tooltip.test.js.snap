// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> Home Page Container renders 1`] = `undefined`;
